src/index.js

```diff
@@ -51,9 +51,9 @@ export default class FacebookTokenStrategy extends OAuth2Strategy {
    * @param {Object} options
    */
   authenticate(req, options) {
-    let accessToken = (req.body && req.body[this._accessTokenField]) || (req.query && req.query[this._accessTokenField]) || (req.headers && req.headers[this._accessTokenField]);
+    let accessToken = (req.body && req.body[this._accessTokenField]) || (req.query && req.query[this._accessTokenField]) || (FacebookTokenStrategy.parseAccessTokenHeader(req.headers, this._accessTokenField, this._authorizationField));
```

Comment:
I agree with your suggestion, it was my original thought as well but I wanted to avoid making too many changes in the library (plus I'm not as familiar with ES6 yet and wasn't sure if there was a better way to handle things).

I will update it.